### DISCUSSION TITLE
support of crontab for endpoints

### DISCUSSION THREAD

**Discussion by umputun** [TwiN/gatus#250] [Ideas]
Currently, there is some way to limit alerts/checks with maintenance windows, however this functionality is limited and does't fit use cases where some boxes/instances go down every night and also still down every weekend. Having the opposite "allowed" time in form of `cron * 9-17 * * 1-5` will allow more flexibility.

_FYI: this [robfig/cron](https://github.com/robfig/cron) library doing a nice job in parsing and scheduling crontab formats_

**Comment by TwiN** [comment]
So to clarify, you'd like to allow the current `maintenance` implementation to use the crontab format, correct?

**Comment by umputun** [comment]
I'd like to allow a single crontab statement to define the opposite to `maintenance`, i.e. what time checks/alerts allowed

UPD: the reason for this - in case of scheduled automation, bringing instances up and down usually defined in a way of "this instance up every workday, 6a to 9p", i.e. similar to crontab style. Inverting it to maintenance rule(s) will require more complex rule with some conditions or multiple rules. In addition, it won't be "maintenance" but rather "instance down" kind of period.

**Comment by TwiN** [comment]
Ah I see - so the crontab statement would be an alternative to the current `endpoints[].interval` parameter.
Sounds like a great idea to me!

**Reply by umputun** [reply]
I think interval is for the duration between checks, this cron is rather how monit does it and calls "service check schedule" 


<img width="1069" alt="tk192-202202-09180627-4leln" src="https://user-images.githubusercontent.com/535880/153311629-b806d771-abc8-4bf1-b30f-6d0b6a3aac66.png">

**Reply by TwiN** [reply]
Ah I see, so you'd like to have the ability to configure the period in which an endpoint is monitored. I assume this would be at the endpoint configuration level too, unlike `maintenance`, which is at the global level.

Yeah that sounds like a great idea, something like `endpoint[].monitoring-window`.
